pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go

```diff
@@ -162,3 +163,35 @@ func getPlatformName(platformType configv1.PlatformType, recorder events.Recorde
 	}
 	return cloudProvider
 }
+
+// IsCloudProviderExternal is used to check whether external cloud provider settings should be used in a component.
+// It checks whether the ExternalCloudProvider feature gate is enabled and whether the ExternalCloudProvider feature
+// has been implemented for the platform.
+func IsCloudProviderExternal(platformType configv1.PlatformType, featureGate *configv1.FeatureGate) (bool, error) {
```

Comment:
When a particular platform goes into GA, the feature gate checks will be ignored and we will just check the platform first and return true, otherwise check the feature gate